### DISCUSSION TITLE
Question: autoNAT implementation

### DISCUSSION THREAD

**Discussion by lucasnethaj** [libp2p/cpp-libp2p#270] [Q&A] [answered]
Hello i got a bit confused because [libp2p/implementations](https://libp2p.io/implementations/) claims that autonat is implemented for cpp. But your readme doesn't mention it and i don't see any references to it in the code.
So is this because i am missing something, or is this feature being worked on by anyone?
Regardless thank you for your time.

**Comment by kamilsa** [comment, marked as answer]
AutoNAT is not implemented in cpp-libp2p

**Comment by srhben** [comment]
emo!!!still not implemented till now...

**Comment by edelmanjm** [comment]
@kamilsa Sorry if this is dumb, but what NAT traversal methods are implemented? [libp2p/implementations](https://libp2p.io/implementations/) still claims that it's just AutoNAT.

**Reply by lucasnethaj** [reply]
I assume none. I made a pr for the website but i guess it's not updated https://github.com/libp2p/website/pull/188
